### ISSUE TITLE
R freezes when running the c-lasso examples through reticulate

### ISSUE THREAD

**Issue by jasonbosch** [rstudio/reticulate#1183]
I am trying to use the R package trac (https://jacobbien.github.io/trac/index.html) which makes use of the python package c-lasso (https://pypi.org/project/c-lasso/). I found that I was unable to get trac to work correctly and I believe that is due to a problem with reticulate as I can work through all the c-lasso examples when running python directly but not when running the examples through reticulate.

I load reticulate and specify the python executable which has all the necessary packages.

```
library(reticulate)
use_python("/usr/bin/python")
```

The first, simple, example ran correctly and finished in 0.885s. When run directly through python it finished in 0.917s.

```
py_run_string("from classo import *")
py_run_string("from numpy import *")
py_run_string("from matplotlib import *")
py_run_string("from scipy import *")
py_run_string("from pandas import *")
py_run_string("from pytest import *")

py_run_string("m, d, d_nonzero, k, sigma = 100, 200, 5, 1, 0.5")
py_run_string("(X, C, y), sol = random_data(m, d, d_nonzero, k, sigma, zerosum=True, seed=1)")

py_run_string("problem = classo_problem(X, y, C) ")
py_run_string("print(problem)")

py_run_string("problem.solve()")
py_run_string("print(problem.solution)")
```

But, when I run the advanced example, R stops responding entirely. It doesn't make a difference if I run it in Rstudio or from the command line.

```
py_run_string("m,  d,  d_nonzero,  k, sigma = 100, 200, 5, 0, 0.5")
py_run_string("(X, C, y), sol = random_data(m, d, d_nonzero, k, sigma, zerosum = True, seed = 4)")
py_run_string("problem                                     = classo_problem(X, y, C)")
py_run_string("problem.formulation.huber                   = True")
py_run_string("problem.formulation.concomitant             = False")
py_run_string("problem.model_selection.CV                  = True")
py_run_string("problem.model_selection.LAMfixed            = True")
py_run_string("problem.model_selection.PATH                = True")
py_run_string("problem.model_selection.StabSelparameters.method = 'max'")
py_run_string("problem.model_selection.CVparameters.seed = 1")
py_run_string("problem.model_selection.LAMfixedparameters.rescaled_lam = True")
py_run_string("problem.model_selection.LAMfixedparameters.lam = .1")

py_run_string("problem.solve()")

py_run_string("print(problem)")
py_run_string("print(problem.solution)")
```

It does not proceed past the problem.solve() command even after several hours. In comparison, when I run the same commands directly in python, they finish in approximately 7s. Even if there were a problem due to R only using a single core, it should still finish in seconds to minutes. As it just stops responding, there is no error message and I am not sure what I can do to solve the problem.

My output for utils::sessionInfo() is:

> R version 4.1.3 (2022-03-10)
> Platform: x86_64-pc-linux-gnu (64-bit)
> Running under: Arch Linux
> 
> Matrix products: default
> BLAS:   /usr/lib/libopenblasp-r0.3.20.so
> LAPACK: /usr/lib/liblapack.so.3.10.0
> 
> locale:
>  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8     LC_MONETARY=en_GB.UTF-8   
>  [6] LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
> [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] reticulate_1.24
> 
> loaded via a namespace (and not attached):
>  [1] Rcpp_1.0.8.3           ape_5.6-2              here_1.0.1             phyloseq_1.36.0        lattice_0.20-45        png_0.1-7             
>  [7] Biostrings_2.60.2      assertthat_0.2.1       rprojroot_2.0.2        foreach_1.5.2          utf8_1.2.2             R6_2.5.1              
> [13] GenomeInfoDb_1.28.4    plyr_1.8.6             stats4_4.1.3           ggplot2_3.3.5          pillar_1.7.0           zlibbioc_1.38.0       
> [19] rlang_1.0.2            rstudioapi_0.13        data.table_1.14.2      vegan_2.5-7            S4Vectors_0.30.2       Matrix_1.4-0          
> [25] splines_4.1.3          stringr_1.4.0          igraph_1.2.11          RCurl_1.98-1.6         munsell_0.5.0          compiler_4.1.3        
> [31] pkgconfig_2.0.3        BiocGenerics_0.38.0    multtest_2.48.0        mgcv_1.8-39            biomformat_1.20.0      tidyselect_1.1.2      
> [37] tibble_3.1.6           GenomeInfoDbData_1.2.6 IRanges_2.26.0         codetools_0.2-18       fansi_1.0.3            permute_0.9-7         
> [43] crayon_1.5.0           dplyr_1.0.8            MASS_7.3-55            rhdf5filters_1.4.0     bitops_1.0-7           grid_4.1.3            
> [49] nlme_3.1-155           jsonlite_1.8.0         gtable_0.3.0           lifecycle_1.0.1        DBI_1.1.2              magrittr_2.0.2        
> [55] scales_1.1.1           cli_3.2.0              stringi_1.7.6          XVector_0.32.0         reshape2_1.4.4         ellipsis_0.3.2        
> [61] generics_0.1.2         vctrs_0.3.8            Rhdf5lib_1.14.2        iterators_1.0.14       tools_4.1.3            ade4_1.7-18           
> [67] Biobase_2.52.0         glue_1.6.2             purrr_0.3.4            parallel_4.1.3         survival_3.2-13        colorspace_2.0-3      
> [73] rhdf5_2.36.0           cluster_2.1.2         

And for  reticulate::py_config():

> python:         /usr/bin/python
> libpython:      /usr/lib/libpython3.10.so
> pythonhome:     //usr://usr
> version:        3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0]
> numpy:          /home/jason/.local/lib/python3.10/site-packages/numpy
> numpy_version:  1.22.3
> 
> NOTE: Python version was forced by use_python function

**Comment by kevinushey** [comment]
Does gdb tell you what the process is stuck doing? FWIW other issues we've seen in the past have been due to incompatible BLASes being mixed in the same process.

**Comment by jasonbosch** [comment]
I eventually solved the problem based on a comment I saw somewhere else. The problem was that I had installed numpy in Python using Pip. Somewhere I read that Arch Linux prefers it's own packages. So I uninstalled numpy with pip and then reinstalled it using Arch's package manager and then it worked perfectly; both through Python itself and when called through reticulate. It even ran faster than before.
